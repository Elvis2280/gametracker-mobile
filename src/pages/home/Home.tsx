import React, { type ReactElement } from 'react'
import SafeViewStyled from '../../components/SafeViewStyled/SafeViewStyled'
import { Avatar, ScrollView, Text, View, XStack, YStack } from 'tamagui'
import useLogin from '../login/hooks/useLogin'
import { useSession } from '../../contex/SessionContext'
import SearchBar from '../../components/SearchBar/SearchBar'
import Filters from '../../components/Filters/Filters'
import { GameCard } from '../../components/GameCard/GameCard'
import AddGame from './components/AddGame'
import { useGetGames } from './hooks/useGetGames'
import { EditGame } from './components/EditGame'

export default function Home(): ReactElement {
  const { handleLogOut } = useLogin()
  const { user } = useSession()
  const { data: games, refetch: getGames } = useGetGames()

  return (
    <SafeViewStyled>
      <YStack height="100%" marginHorizontal={'$4'}>
        <ScrollView
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={[0]}
        >
          {/* Header */}
          <XStack
            backgroundColor={'$blue1'}
            paddingVertical={'$2'}
            alignItems={'center'}
            height={'$6'}
          >
            <Text
              flex={1}
              fontSize={'$8'}
              fontWeight={'bold'}
              textAlign={'center'}
            >
              Gametracker
            </Text>
            <Avatar
              position={'absolute'}
              right={0}
              onPress={handleLogOut}
              circular
              backgroundColor={'$blue4'}
            >
              <Text fontSize={'$6'} fontWeight={'bold'}>
                {user?.username.charAt(0)}
              </Text>
            </Avatar>
          </XStack>
          {/* body */}
          <View
            position={'relative'}
            flex={1}
            marginTop={'$6'}
            paddingBottom={'$2'}
          >
            {/* search Section */}
            <XStack space={'$4'}>
              <View flex={1}>
                <SearchBar
                  handleSearch={() => {}}
                  placeholder={'Search in your game list'}
                />
              </View>
              <Filters />
            </XStack>

            {/* Games Cards */}
            <View marginTop={'$4'}>
              <Text fontSize={'$6'} fontWeight={'bold'} paddingBottom={'$2'}>
                Games List
              </Text>
              <YStack space={'$4'}>
                {games?.map((game, i) => {
                  return (
                    <GameCard
                      editGameButton={
                        <EditGame handleOnSuccess={getGames} game={game} />
                      }
                      key={i}
                      game={game}
                    />
                  )
                })}
              </YStack>
            </View>
          </View>
        </ScrollView>

        <AddGame onSuccessCallback={getGames} />
      </YStack>
    </SafeViewStyled>
  )
}
