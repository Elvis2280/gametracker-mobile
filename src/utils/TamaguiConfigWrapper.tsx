import React, { type ReactElement } from 'react'
import { TamaguiProvider, Theme } from 'tamagui'
import tamaguiConfig from '../../tamagui.config'
import { QueryClient, QueryClientProvider } from 'react-query'

interface Props {
  children: ReactElement
}

export default function TamaguiConfigWrapper({
  children
}: Props): ReactElement {
  const queryClient = new QueryClient()
  return (
    <QueryClientProvider client={queryClient}>
      <TamaguiProvider config={tamaguiConfig}>
        <Theme name={'blue_alt1'}>{children}</Theme>
      </TamaguiProvider>
    </QueryClientProvider>
  )
}
