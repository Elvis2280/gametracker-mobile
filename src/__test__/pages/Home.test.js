import React from 'react'
import Home from '../../pages/home/Home'
import TamaguiConfigWrapper from '../../utils/TamaguiConfigWrapper'
import { render } from '@testing-library/react-native'

describe('Home', () => {
  it('renders correctly', () => {
    const rendered = render(
      <TamaguiConfigWrapper>
        <Home />
      </TamaguiConfigWrapper>
    )
    expect(rendered).toMatchSnapshot()
  })
})
