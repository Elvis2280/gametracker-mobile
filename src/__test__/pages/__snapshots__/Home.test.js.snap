// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "hsl(212, 35.0%, 9.2%)",
      "minHeight": "100%",
    }
  }
>
  <RCTSafeAreaView>
    <View
      style={{}}
    >
      <Text
        fontFamily="unset"
        style={
          {
            "color": "hsl(206, 98.0%, 95.8%)",
            "fontFamily": "Inter",
          }
        }
        suppressHighlighting={true}
      >
        Home
      </Text>
      <View
        accessibilityRole="button"
        cursor="pointer"
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "hsl(213, 71.2%, 20.2%)",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 9,
            "borderBottomRightRadius": 9,
            "borderBottomWidth": 1,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 1,
            "borderRightColor": "transparent",
            "borderRightWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 9,
            "borderTopRightRadius": 9,
            "borderTopWidth": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": 44,
            "justifyContent": "center",
            "paddingLeft": 18,
            "paddingRight": 18,
          }
        }
      >
        <Text
          fontFamily="unset"
          style={
            {
              "color": "hsl(206, 98.0%, 95.8%)",
              "fontFamily": "Inter",
            }
          }
          suppressHighlighting={true}
        >
          Logout
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
